pkg/kube/cr_manager.go

```diff
@@ -380,6 +381,14 @@ func (m *crManager) cleanupNamespace(ctx context.Context) (err error) {
 	if err != nil && !errors.IsNotFound(err) {
 		return
 	}
+	for {
+		time.Sleep(2 * time.Second)
+		ns, err := m.clientset.CoreV1().Namespaces().Get(ctx, NamespaceStarboard, meta.GetOptions{})
+		if errors.IsNotFound(err) {
+			klog.V(3).Infof("Starboard deleted: %v", ns)
+			break
+		}
+	}
```

Comment:
```suggestion
	for {
		select {
		// This case controls the polling interval (every 2 seconds)
		case <-time.After(2 * time.Second):
			_, err := m.clientset.CoreV1().Namespaces().Get(ctx, NamespaceStarboard, meta.GetOptions{})
			if errors.IsNotFound(err) {
				klog.V(3).Infof("Deleted Namespace %v", NamespaceStarboard)
				return nil
			}
		// This case caters for timeout (stop polling after 30 seconds)
		case <-time.After(30 * time.Second):
			return fmt.Errorf("deleting namespace timed out")
		}
	}
```

Reply:
The for loop with time.Sleep function would do the job as you implemented it. However, I'd suggest adding the timeout case to avoid very long polling

Reply:
Thanks!! Nice suggestion !